autogluon/task/tabular_prediction/predictor.py

```diff
@@ -361,10 +361,14 @@ def transform_features(self, dataset=None, model=None, base_models=None):
                 For non-bagged mode predictors:
                     The dataset used when not specified is the validation set.
                     This can either be an automatically generated validation set or the user-defined `tuning_data` if passed during fit().
+                    If all parameters are unspecified, then the output is equivalent to `predictor.load_data_internal(dataset='val', return_X=True, return_y=False)[0]`.
```

Comment:
why is the validation set the default? I think I'm missing what the use-case would be there.   I thought the main use case would be for users to train 3rd party models on the same data representation, in which case the training set would be the default?

Reply:
It must default to ‘val’ for non-bagged because it would crash if it was ‘train’ and the user asked for weighted ensemble input (because it doesn’t exist)